### PR TITLE
Steven Yau

### PR DESCRIPTION
**Pull request by akorolczuk**
Readme here: https://github.com/yaustar/BCoTankGame/blob/master/readme.md

### CODE REVIEW

**Comment by akorolczuk** [comment]
I think it looks ok, a bit worried about MonoBehaviours used for all game objects but at least there's reasonable structure between them. What do you think @Moniraego? I think it's best if you interview him since I know him personally.

File: StevenYau/Assets/TankGame/Data/LeaderboardTableData.cs [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+        
+        public List<ScoreEntryData> table = new List<ScoreEntryData>();
+
+        
+        public bool IsHighScore(int score) {
+            if (score <= 0) {
+                return false;
+            }
+            
+            if (table.Count == 0 || table.Count < MAX_ENTRIES) {
```
Comment (Moniraego) [inline StevenYau/Assets/TankGame/Data/LeaderboardTableData.cs]: Redundant condition

File: StevenYau/Assets/TankGame/Scripts/Ui/UpdateUiIconCounterFromIntVar.cs [hunk 2]
```diff
@@ -0,0 +1,70 @@
+﻿using System;
+using System.Collections;
+using System.Collections.Generic;
+using NaughtyAttributes;
+using UnityEngine;
+
+namespace TankGame {
+    public class UpdateUiIconCounterFromIntVar : MonoBehaviour {
```
Comment (Moniraego) [inline StevenYau/Assets/TankGame/Scripts/Ui/UpdateUiIconCounterFromIntVar.cs]: Sort of a weird class name

File: StevenYau/Assets/TankGame/Scripts/Ui/UpdateUiIconCounterFromIntVar.cs [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+        private int _lastFrameValue = Int32.MaxValue;
+        
+
+        private void Awake() {
+            _iconPool = new ObjectPool(_iconPrefab, _iconsPoolContainerObject, 5);
+        }
+
+        
+        void Start() {
+            // Very broken, can't unbind the listener!
```
Comment (Moniraego) [inline StevenYau/Assets/TankGame/Scripts/Ui/UpdateUiIconCounterFromIntVar.cs]: Hmmm

File: StevenYau/Assets/TankGame/Scripts/World/GiveDamage.cs [hunk 4]
```diff
@@ -0,0 +1,54 @@
+﻿using System;
+using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+
+
+namespace TankGame {
+    public class GiveDamage : MonoBehaviour {
```
Comment (Moniraego) [inline StevenYau/Assets/TankGame/Scripts/World/GiveDamage.cs]: Again, this should be a function name.

**Pull request closed without merging**